File: php/return_top_hits.php
<?php

# This script makes a GET request to the /searches endpoint on the Analytics REST API - https://www.algolia.com/doc/rest-api/analytics/. 
# To get the top 1000 searches over the last 7 days.

require __DIR__.'/vendor/autoload.php';

use Algolia\AlgoliaSearch\Api\AnalyticsClient;

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

# Algolia client credentials
$ALGOLIA_APP_ID = $_ENV['ALGOLIA_APP_ID'];
$ALGOLIA_API_KEY = $_ENV['ALGOLIA_API_KEY'];
$ALGOLIA_INDEX_NAME = $_ENV['ALGOLIA_INDEX_NAME'];

# Get application region
# https://www.algolia.com/doc/libraries/sdk/methods/analytics#php

$ALGOLIA_APPLICATION_REGION = "us";

$client = AnalyticsClient::create($ALGOLIA_APP_ID, $ALGOLIA_API_KEY, $ALGOLIA_APPLICATION_REGION);

$response = $client->getTopSearches($ALGOLIA_INDEX_NAME);

# Returning 1000 Top Searches
print("1000 Top Searches:\n");
var_dump($response);

# Format the resulting json string
$formatted_result = json_encode($response, JSON_PRETTY_PRINT);

# Write json to file
if (file_put_contents("{$ALGOLIA_INDEX_NAME}_top_1000_searches.json", $formatted_result))
    echo "JSON file created successfully...\n";
else
    echo "Oops! Error creating json file...\n";

File: java/src/main/java/ReturnTopHits.java
import java.io.File;
import java.io.IOException;
import java.util.concurrent.ExecutionException;

import com.algolia.api.AnalyticsClient;
import com.algolia.config.*;
import com.algolia.model.analytics.GetTopSearchesResponse;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;

import io.github.cdimascio.dotenv.Dotenv;

public class ReturnTopHits {
    
    public static void main(String[] args) throws ExecutionException, InterruptedException, IOException {
        ReturnTopHits.run();
    }

    public static void run() throws ExecutionException, InterruptedException, IOException {
    
        Dotenv dotenv = Dotenv.configure().load();

        // Pick a region for your analytics
        // us for the United States or de for Europe

        String appID = dotenv.get("ALGOLIA_APP_ID");
        String apiKey = dotenv.get("ALGOLIA_API_KEY");
        String indexName = dotenv.get("ALGOLIA_INDEX_NAME");
        String analyticsRegion = "us";

        // Start the API client
        // https://www.algolia.com/doc/libraries/sdk/methods/analytics
        // Initialize the client with your application region, eg. analytics.ALGOLIA_APPLICATION_REGION
        try (AnalyticsClient client = new AnalyticsClient(appID, apiKey, analyticsRegion)) {
            // Initialize ObjectMapper
            ObjectMapper objectMapper = new ObjectMapper();
            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);

            RequestOptions options = new RequestOptions();
            options.addExtraQueryParameters("limit",1000);
            GetTopSearchesResponse topSearchResp = client.getTopSearches(
                indexName,
                null,   // clickAnalytics
                null,   // revenueAnalytics
                null,   // startDate
                null,   // endDate
                null,   // orderBy
                null,   // direction
                1000,
                null,
                null    // tags
            );

            System.out.println(topSearchResp);

            File recordsFile = new File(indexName + "__top_1000_searches.json");
            objectMapper.writeValue(recordsFile, topSearchResp); 
        }
    }
}